### ISSUE TITLE
Chat with page suggestion

### ISSUE THREAD

**Issue by priyavrat-misra** [n4ze3m/page-assist#56]
Although it generally works well, it deviates from the answers due to the presence of extraneous html components in the context.
I was wondering if it was feasible for the extension to evaluate only the element that we explicitly select using the inspect tool, so that the results could be generated with accurate context.

**Comment by n4ze3m** [comment]
I like this idea. I will add a feature like the inspect tool in Chrome DevTools, which allows you to select the content you want to chat about.

**Comment by lboklin** [comment]
It appears that sometimes only CSS is embedded as the model can not discern any content and will only talk about styling and code. Is it possible to attempt some document conversion by default, similar to firefox's reader mode?

**Comment by n4ze3m** [comment]
Yes, I'm adding Mozilla's readability to the chat with website mode

**Comment by oatmealm** [comment]
I don't know if you want to add 3rd party services to this but solutions like jina exist for this purpose and don't require the user to get and api key. Admittedly this is tricky because it could change in the future I guess.

**Comment by n4ze3m** [comment]
the latest version of the Page Assist website parser works similarly to the Jina one. It needs a few more improvements, but overall, it's the same

**Comment by GabeDuarteM** [comment]
@n4ze3m  One thing I was looking for when I installed the extension, and that could be related to this feature request, was something similar to brave's `leo ai`, where you can select a piece of text and then prompt the ai something regarding it.

![image](https://github.com/user-attachments/assets/6e36984c-0c1c-4d04-a39d-b7a613c232c4)

It lists a few presets of options, like summarize, explain, etc, but ideally it would be good if custom ones could be added (probably a list of `title` and `system prompt` to use when that option is selected), then when you select a text (could be a paragraph, or the whole article) and click on an option, it would send that text as a prompt, and together with the system prompt (or something similar, like just prepending the selected text with something like `Summarize the following text:` or `Translate the following text`) I guess it would generate the desired outcome :thinking: 

Also, let me know if this is better off as a separate issue, I just thought this might be simpler to implement as it wouldn't require a `inspect dom` tool, and at the same time would achieve the same goal, but with more flexibility since you would be able to pre-define custom prompts as well

**Comment by n4ze3m** [comment]
Hey guys, the latest version supports a Leo-like feature. Please note that you can edit the prompt from the 'Manage Prompt' section

![image](https://github.com/user-attachments/assets/e42456fe-8351-44da-8da3-5e433ba4ea1f)
![InnSDdu77P](https://github.com/user-attachments/assets/ea6c15cf-2383-4a10-ab45-749a25d1a12c)

**Comment by oatmealm** [comment]
That looks amazing. I'm not seeing an update though? Should I install manually, if it not on the store yet?

**Comment by oatmealm** [comment]
Ok sorry, it's on text selection specifically :) Wow. Works for me, and the prompt seems to be bang on! Very impressed!

![image](https://github.com/user-attachments/assets/d3567dd4-fa71-4104-a533-a58b9b4caebb)
